app/jobs/close_patients_job.rb

```diff
@@ -30,8 +31,25 @@ def perform(*_args)
         patient[:monitoring_reason] = 'Completed Monitoring (system)'
       end
 
-      # Send closed email to patient if they are a reporter
-      PatientMailer.closed_email(patient).deliver_later if patient.save! && patient.email.present? && patient.self_reporter_or_proxy?
+      # Determine if the patient's jurisdiction allows automated closed notifications
+      jurisdiction_send_close[patient.jurisdiction_id] = patient.jurisdiction.send_close unless jurisdiction_send_close.key? patient.jurisdiction_id
+      send_close = jurisdiction_send_close[patient.jurisdiction_id]
```

Comment:
I'll write a benchmark and check back in here

Reply:
```
Calculating -------------------------------------
         rails_cache    115.327  (±12.1%) i/s -    567.000  in   5.005704s
          hash_cache    329.727  (± 2.1%) i/s -      1.650k in   5.006641s

Comparison:
          hash_cache:      329.7 i/s
         rails_cache:      115.3 i/s - 2.86x  (± 0.00) slower
```

Source Code for Benchmark:
```
# frozen_string_literal: true

require 'benchmark/ips'

module Benchmarking
  @patients = Patient.all

  def rails_cache
    @patients.each do |patient|
      send_close = patient.jurisdiction.send_close
    end
  end

  def hash_cache
    jurisdiction_send_close = {} # { jurisdiction_id: send_close, ... }
    @patients.each do |patient|
      jurisdiction_send_close[patient.jurisdiction_id] = patient.jurisdiction.send_close unless jurisdiction_send_close.key? patient.jurisdiction_id
      send_close = jurisdiction_send_close[patient.jurisdiction_id]
    end
  end

  def benchmark
    puts "Benchmarking with #{Patient.count} patients..."
    Benchmark.ips do |x|
      x.report('rails_cache') { rails_cache }
      x.report('hash_cache') { hash_cache }
      x.compare!
    end
  end
end
```

Reply:
Suggestion:
L13: `jurisdiction_send_close = Jurisdiction.where(id: eligible.pluck(:jurisdiction_id)).pluck(:id, :send_close).to_h`

How does this compare? We have all the data to create the data structure immediately

Reply:
Tested on a 500k database.

It turns out that plucking all of the patient ids is expensive with that query, but I tried just plucking ALL jurisdiction ids and send_close and it is actually the fastest (even with 3000+ jurisdictions)

```
Comparison:
       all_jur_cache:      109.1 i/s
          hash_cache:        3.7 i/s - 29.59x  (± 0.00) slower
      patient_jur_cache:        0.5 i/s - 232.67x  (± 0.00) slower

hash_cache -> same as current code
patient_jur_cache -> your proposed method
all_jur_cache -> similar to your proposed method BUT just gets all jurisdiction data
```

Reply:
I think the benchmark code is flawed because you wouldn't be calling `Jurisdiction.where` or `Jurisdiction.all`every single time. Only once per job. Unlike the loops in the first two methods which do have a more significant cost. I'm good with `Jurisdiction.pluck(:id, :send_close).to_h`. I had kind of written that off because we were getting more data than necessary and I just assumed it would be less CPU performant.